sklearn/cluster/optics_.py

```diff
@@ -362,25 +380,33 @@ def fit(self, X, y=None):
 
         nbrs = NearestNeighbors(n_neighbors=self.min_samples,
                                 algorithm=self.algorithm,
-                                leaf_size=self.leaf_size, metric=self.metric,
-                                metric_params=self.metric_params, p=self.p,
+                                leaf_size=self.leaf_size,
+                                metric=self.metric,
+                                metric_params=self.metric_params,
+                                p=self.p,
                                 n_jobs=self.n_jobs)
 
         nbrs.fit(X)
-        self.core_distances_[:] = nbrs.kneighbors(X,
-                                                  self.min_samples)[0][:, -1]
+        self.core_distances_[:] = nbrs.kneighbors(
+            X, self.min_samples)[0][:, -1]
 
         self.ordering_ = self._calculate_optics_order(X, nbrs)
 
-        indices_, self.labels_ = _extract_optics(self.ordering_,
-                                                 self.reachability_,
-                                                 self.maxima_ratio,
-                                                 self.rejection_ratio,
-                                                 self.similarity_threshold,
-                                                 self.significant_min,
-                                                 self.min_cluster_size,
-                                                 self.min_maxima_ratio)
+        if self.extract_method == 'fancy':
+            indices_, labels_ = _extract_optics(
+                self.ordering_,
+                self.reachability_,
+                self.maxima_ratio,
+                self.rejection_ratio,
+                self.similarity_threshold,
+                self.significant_min,
+                self.min_cluster_size,
+                self.min_maxima_ratio)
+        elif self.extract_method == 'dbscan':
+            indices_, labels_ = self.extract_dbscan(self.eps)
```

Comment:
It might be nicer if this were a function so that we can see what it is a function of?

Reply:
I'm not sure if I understand to what "this" is referring. Do you want an `_extract(extract_method, extract_params:dict)` outside the class for instance?